PeopleCounterApi/Controllers/PeopleCounterController.cs

```
@@ -62,57 +51,34 @@ public async Task<IActionResult> GetShops(string shopId, DateTime from, DateTime
-
-            var thePeopleCountersCollection = _mapper.Map<IEnumerable<PeopleCounter>>(peopleCounters);
+            if (peopleCountersVm == null)
+                return BadRequest("peopleCountersVm cannot be null");
 
-            IEnumerable<PeopleCounter> peopleCounterToAdd = thePeopleCountersCollection.ToList();
-            await _peopleCounterService.AddPeopleCounters(peopleCounterToAdd);
+            var peopleCounters = _mapper.Map<IEnumerable<PeopleCounter>>(peopleCountersVm);
+            await _peopleCounterService.AddPeopleCounters(peopleCounters);
 
-            if (!peopleCounterToAdd.Any())
+            if (!peopleCounters.Any())
             {
                 return BadRequest("Failed to create PeopleCounterCollection");
             }
 
-            peopleCounters = _mapper.Map<IEnumerable<PeopleCounterGetViewModel>>(thePeopleCountersCollection);
-            _messageSender.Send(PeopleCounterEvent.PeopleCounterAdded, peopleCounters);
-
-            return Ok(peopleCounters);
+            return Ok(peopleCountersVm);
         }
 
-        [HttpDelete("Delete/{id}")]
-        [ProducesResponseType(typeof(int), (int)HttpStatusCode.OK)]
+        [HttpDelete("{id}")]
+        [ProducesResponseType(typeof(int), (int) HttpStatusCode.OK)]
         public async Task<IActionResult> Delete(int id)
         {
             int deletedCount = 0;
 
-            var peopleCounter = await _peopleCounterService.GetPeopleCounterById(id);
-            if (peopleCounter == null)
-            {
-                return NotFound($"PeopleCounter Id={id} not found");
-            }
-
-            try
-            {
-                deletedCount = await _peopleCounterService.DeletePeopleCounterById(id);
-            }
-            catch (ArgumentException)
-            {
-                return NotFound($"PeopleCounter Id={id} not found");
-            }
-
+            deletedCount = await _peopleCounterService.DeletePeopleCounterById(id);
```

Also here I think the DeletePeopleCounterById should throw an exception if it fails. Then we do not need to check return values as it will be handled by the middleware.

---

not sure if this is what you mean, but i created the add update delete exceptions to report a bad request.